LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js

```diff
@@ -9,59 +9,54 @@ function logDebug(callback)
 
 function pause(duration) {
     return new Promise(resolve => {
-        setTimeout(() => {
-            resolve();
-        }, duration);
+        setTimeout(resolve, duration);
     });
 }
 
-function dispatchMouseActions(actions, pointerType)
+async function dispatchMouseActions(actions, pointerType)
 {
     if (!window.eventSender)
-        return Promise.reject(new Error("window.eventSender is undefined."));
+        throw new Error("window.eventSender is undefined.");
 
-    return new Promise(resolve => {
-        setTimeout(async () => {
-            eventSender.dragMode = false;
-
-            for (let action of actions) {
-                switch (action.type) {
-                case "pointerMove":
-                    let origin = { x: 0, y: 0 };
-                    if (action.origin instanceof Element) {
-                        const bounds = action.origin.getBoundingClientRect();
-                        logDebug(() => `${action.origin.id} [${bounds.left}, ${bounds.top}, ${bounds.width}, ${bounds.height}]`);
-                        origin.x = bounds.left + 1;
-                        origin.y = bounds.top + 1;
-                    }
-                    logDebug(() => `eventSender.mouseMoveTo(${action.x + origin.x}, ${action.y + origin.y})`);
-                    eventSender.mouseMoveTo(action.x + origin.x, action.y + origin.y, pointerType);
-                    break;
-                case "pointerDown":
-                    logDebug(() => `eventSender.mouseDown()`);
-                    eventSender.mouseDown(action.button, [], pointerType);
-                    break;
-                case "pointerUp":
-                    logDebug(() => `eventSender.mouseUp()`);
-                    eventSender.mouseUp(action.button, [], pointerType);
-                    break;
-                case "pause":
-                    logDebug(() => `pause(${action.duration})`);
-                    await pause(action.duration);
-                    break;
-                default:
-                    return Promise.reject(new Error(`Unknown action type "${action.type}".`));
-                }
+    await setTimeout(new Promise(resolve => resolve()), 0);
```

Comment:
Does this work as expected? I think it should be the other way round:

`await new Promise(resolve => setTimeout(resolve, 0));`

Reply:
oh whoops. Good catch!

Reply:
This is why I'd like to add more TypeScript annotations... and hopefully get us using a bit more TypeScript. It catches silly mistakes like this in the code editor.